test/adaptive_load/scoring_function_test.cc

```diff
@@ -0,0 +1,231 @@
+#include "external/envoy/source/common/config/utility.h"
+
+#include "adaptive_load/scoring_function_impl.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+
+TEST(BinaryScoringFunctionConfigFactoryTest, GeneratesEmptyConfigProto) {
+  ScoringFunctionConfigFactory& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<ScoringFunctionConfigFactory>(
+          "nighthawk.binary");
+  Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig expected_config;
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
```

Comment:
I learned that with proto3 default values may not get serialised. I'm not 100% sure that equivalence here guarantees actual equivalence with respect to set/not-set/default-value. Therefore, to make sure no edge cases slip past, maybe it makes sense to also double down with `MessageUtil()(message1, message1)`. This won't produce a helpful test message, so I'd put it after comparing the debug strings (which should produce a nice diff), maybe in a helper method.

Reply:
Done, filed #433 to create the helper once all these small PRs are done.